src/hades_types/config.py

```
@@ -0,0 +1,193 @@
+        return cls(**config_data)
+    
+    def to_yaml_files(self, config_dir: str) -> None:
+        """Save configuration to YAML files."""
+        import yaml
+        from pathlib import Path
+        
+        config_path = Path(config_dir)
+        try:
+            config_path.mkdir(exist_ok=True)
+        except OSError as e:
+            raise IOError(f"Failed to create config directory {config_dir}: {e}")
```

Ruff (0.11.9)

Within an `except` clause, raise exceptions with `raise ... from err` or `raise ... from None` to distinguish them from errors in exception handling

(B904)